adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -691,6 +735,11 @@ private void waitingRequestOnError(final CallbackHandler handler, final Authenti
             if (waitingRequest != null && waitingRequest.getDelegate() != null) {
                 Logger.v(TAG, "Sending error to callback"
                         + mAuthContext.getCorrelationInfoFromWaitingRequest(waitingRequest));
+                waitingRequest.getAPIEvent().setWasApiCallSuccessful(false);
+                waitingRequest.getAPIEvent().setCorrelationId(
+                        waitingRequest.getRequest().getCorrelationId().toString());
```

Comment:
Why we're adding the correlation id? Shouldn't it be already set when you start the event?